packages/squiggle-lang/__tests__/Reducer/Reducer_externalBindings_test.res

```diff
@@ -0,0 +1,60 @@
+open Jest
+open Reducer_TestHelpers
+
+describe("Parse for Bindings", () => {
+  testParseOuterToBe("x", "Ok((:$$bindExpression (:$$bindings) :x))")
+  testParseOuterToBe("x+1", "Ok((:$$bindExpression (:$$bindings) (:add :x 1)))")
+  testParseOuterToBe(
+    "y = x+1; y",
+    "Ok((:$$bindExpression (:$$bindStatement (:$$bindings) (:$let :y (:add :x 1))) :y))",
+  )
+})
+
+describe("Parse for Bindings", () => {
+  testParsePartialToBe(
+    "x",
+    "Ok((:$$bindExpression (:$$bindStatement (:$$bindings) :x) (:$exportVariablesExpression)))",
+  )
+  testParsePartialToBe(
+    "y=x",
+    "Ok((:$$bindExpression (:$$bindStatement (:$$bindings) (:$let :y :x)) (:$exportVariablesExpression)))",
+  )
+  testParsePartialToBe(
+    "y=x+1",
+    "Ok((:$$bindExpression (:$$bindStatement (:$$bindings) (:$let :y (:add :x 1))) (:$exportVariablesExpression)))",
+  )
+  testParsePartialToBe(
+    "y = x+1; z = y",
+    "Ok((:$$bindExpression (:$$bindStatement (:$$bindStatement (:$$bindings) (:$let :y (:add :x 1))) (:$let :z :y)) (:$exportVariablesExpression)))",
+  )
+})
+
+describe("Eval with Bindings", () => {
+  testEvalBindingsToBe("x", list{("x", ExpressionValue.EvNumber(1.))}, "Ok(1)")
```

Comment:
To be clear; is list{()} basically the local environment that's being passed down? If so, I could see it being good to have a custom type for this, with list as an internal thing. Like,
``reducerEnv("x", ExpressionValue.EvNumber(1.))``

Reply:
Guys, it is only the testing interface.  Sorry for that. It is actually JavaScript record in/out for the actual eval function. 
JavaScript record corresponds to Js.Dict<expressionValue> . However I need to construct the Js.Dict from a list. I cannot write it directly like in JavaScript. So the testing interface is getting a list and converting it. In rescript, variant records could be used to have a record notation but they are totally typeless and I don't want it.

Reply:
Couldn't you do something like create a custom type for this, still?

Then you'd call this like,

```
   testEvalBindingsToBe("x", ReducerEnv.make(list{("x", ExpressionValue.EvNumber(1.))}), "Ok(1)") 
```  

I don't think the javascript should be a big deal, because we'll be wrapping that into a more custom Typescript interface/library anyway.